modules/ding_ddbasic/modules/ding_ddbasic_opening_hours/ding_ddbasic_opening_hours.module

```diff
@@ -0,0 +1,166 @@
+    'page arguments' => array(3, 4, 5),
+    'delivery callback' => 'ajax_deliver',
+    'access callback' => TRUE,
+    'type' => MENU_CALLBACK,
+    'file' => 'ding_ddbasic_opening_hours.pages.inc',
+  );
+  $items['ding-ddbasic/opening-hours/week/%node/%/%/%']['page arguments'][] = 6;
+
+  return $items;
+}
+
+/**
+ * Implements hook_init().
+ */
+function ding_ddbasic_opening_hours_init() {
+  $item = menu_get_item();
+  if ($item['path'] === 'node/%') {
+    $node = menu_get_object();
+    if ($node->type === 'ding_library' && variable_get('ding_ddbasic_opening_hours_expand_on_library', FALSE)) {
+      drupal_add_js(array('ding_ddbasic_opening_hours_expand_on_library' => TRUE), 'setting');
+    }
+  }
+}
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function ding_ddbasic_opening_hours_ctools_plugin_directory($module, $plugin) {
+  if ($module == 'ctools' && !empty($plugin)) {
+    return "plugins/$plugin";
+  }
+}
+
+/**
+ * Implements hook_theme().
+ *
+ * - All opening hours theme.
+ */
+function ding_ddbasic_opening_hours_theme($existing, $type, $theme, $path) {
+  return array(
+    'ding_ddbasic_opening_hours_all' => array(
+      'file' => 'ding_ddbasic_opening_hours.theme.inc',
+      'variables' => array(
+        'today' => NULL,
+      ),
+    ),
+    'ding_ddbasic_opening_hours_week' => array(
+      'file' => 'ding_ddbasic_opening_hours.theme.inc',
+      'variables' => array(
+        'node' => NULL,
+        'categories' => array(),
+        'timespan' => array(),
+      ),
+    ),
+  );
+}
+
+/**
+ * Turn a structrued "opening hours" array into a table theme render array.
+ */
+function ding_ddbasic_opening_hours_table($title, $categories, $structured) {
+  $table = array(
+    '#theme' => 'table',
+    '#header' => array_merge(array($title), $categories),
+    '#rows' => array(),
+    '#attributes' => array('class' => array('opening-hours-table')),
+    '#sticky' => FALSE,
+  );
+
+  foreach ($structured as $label => $row) {
+    if (!isset($row['name'])) {
+      $cols = array($label);
+    }
+    else {
+      $cols = array($row['name']);
+    }
+
+    foreach ($categories as $category_weight => $header) {
+      if (!empty($row['cols'][$category_weight])) {
+        $col = array(
+          'data' => $row['cols'][$category_weight],
+          'data-label' => $header,
+        );
+      }
+      else {
+        $col = array('class' => array('empty'));
+      }
+
+      $cols[] = $col;
+    }
+
+    $table['#rows'][] = array('data' => $cols) + $row['extra'];
+  }
+
+  return $table;
+}
+
+/**
+ * Get name of an opening hours category tid.
+ */
+function ding_ddbasic_opening_hours_get_category_name($category_tid) {
+  $name = &drupal_static(__FUNCTION__, array());
+
+  if (!isset($name[$category_tid])) {
+    if ($category_tid === NULL) {
+      $name[$category_tid] = t('Opening hours');
+    }
+    else {
+      $name[$category_tid] = taxonomy_term_load($category_tid)->name;
```

Comment:
In my head, "tid" refereces to the taxonomy term ID in Drupal, like you have "nid" for nodes.

It's not really something you must fix, I was just curious if it was a slip up, as maybe the code else where would expect a real tid, instead of a string?

Also, because you're using a translated string as the tid, it'll change for every language. Dont you want the tid to be language neutral?

About the last thing, the translation string, i understand what you mean now and i can accept it 👍